### ISSUE TITLE
Execution optimizations are disabled because of undeclared dependency on kspCommonMainKotlinMetadata

### ISSUE THREAD

**Issue by eygraber** [google/ksp#975]
Using ksp 1.6.21-1.0.5 I get a warning in my kotlin compile tasks:

```
Execution optimizations have been disabled for task ':core:compileDebugKotlinAndroid' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/home/eli/workspace/galgal/core/build/generated/ksp/metadata/commonMain/kotlin'. Reason: Task ':core:compileDebugKotlinAndroid' uses this output of task ':core:kspCommonMainKotlinMetadata' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.4.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
```

I can work around it by doing:

```
tasks.withType<KotlinCompile>().configureEach {
  if(name != "kspCommonMainKotlinMetadata") {
    dependsOn("kspCommonMainKotlinMetadata")
  }
}
```

**Comment by neetopia** [comment]
I wonder if this is actually breaking anything other than producing a warning?

**Comment by eygraber** [comment]
It doesn't break anything in the sense that it can't build, but I believe it executes in a non optimal way (as the warning indicates)

**Comment by ting-yuan** [comment]
Looks like the output of `kspCommonMainKotlinMetadata` is wired to the target source set manually by user. This use case will be supported by a new project model in 1.0.8.

**Comment by ting-yuan** [comment]
Fixed in e81d01f80e8da40542f22e8cbeec724aa22d6062.
